pkg/storing/events/events.go

```diff
@@ -137,28 +145,79 @@ func (s *Storage) LatestHeight(node string) (int, error) {
 		return 0, errors.Wrapf(err, "failed to get latest height for node '%s'", node)
 	}
 	if h == 0 {
-		return 0, errors.Errorf("no full statements for node '%s'", node)
+		return 0, errors.Wrapf(NoFullStatementError, "no full statement at latest height for node '%s'", node)
 	}
 	return h, nil
 }
 
-var BigHeightDifference = errors.New("The height difference between nodes is more than 10")
-var StorageIsNotReady = errors.New("The storage has not collected enough statements for status")
-
-func (s *Storage) FindAllStatehashesOnCommonHeight(nodes []string) ([]entities.NodeStatement, error) {
-	var heightDifference = 10
-
-	minHeight, maxHeight := math.MaxInt, 0
+func (s *Storage) StatementAtHeight(node string, height int) (entities.NodeStatement, error) {
+	pattern := newStatementKey(node, "*")
+	var st entities.NodeStatement
+	notFound := false
+	err := s.viewByKeyPatternWithDescendKeys(pattern, func(s *entities.NodeStatement) bool {
+		st = *s
+		if st.Height < height {
```

Comment:
We also need to handle rollback cases. For example, change the `if` condition like this:
```go
const maxHeightThreshold = 100
if st.Height < height && height - st.Height > maxHeightThreshold {
    // snip
}
```

Reply:
This function is not supposed to do that. It looks for a value at an exact height. There are other checks for rollback cases